tests/Avalonia.Controls.UnitTests/TransitioningContentControlTests.cs

```diff
@@ -167,7 +167,7 @@ public void New_Transition_Should_Be_Started_If_Content_Changes_While_Running()
 
                 Assert.Same(presenter2, fromPresenter);
                 Assert.Same(target.Presenter, toPresenter);
-                Assert.Equal("bar", fromPresenter.Content);
+                Assert.Equal(null, fromPresenter.Content); //We explicitly nullify old Content as soon as we are done with it so that it can be safely disposed by any ViewModel observing OldContent property
```

Comment:
As mentioned on previous comments, this doesn't seem right: the transition has just started and hasn't been complete, the content shouldn't be null already. These tests shouldn't be changed.

Reply:
I believe this is correct because we are tackling the case of Content being changed _while the transition is running_, and so we therefore have 3 "Contents" the "From" and "To" Contents in the running transition, and the newest Content which has just been set.

So the logic is, we should ignore the current transition (it should be cancelled with cancellation token in such a case), and then proceed with the newest Content's logic (i.e. we set the "From"'s content of the running transition to null (the oldest of them all)).

Reply:
The main issue I was trying to tackle is how to properly notify of OldContent which arises when new content is being set _while the transition is still running_ (i.e. 3 "Contents" situation).

Reply:
No, this isn't correct: just try it in the `ControlCatalog` and see the old content missing if you try to change the content while the transition is in progress. e.g transitioning from A to B then B to C (while the first transition A to B hasn't finished) gives a transition from null to C.

I believe you're overthinking this a little :) The current logic before this PR already correctly releases the old content and doesn't have to change. The only thing missing is to notify of this old content whenever that happens, that is when any `ContentPresenter.Content` changes.